Source/diablo.h

```diff
@@ -41,8 +41,8 @@ int __fastcall diablo_init_menu(int a1, int bSinglePlayer);
 void __fastcall run_game_loop(int uMsg);
 void __fastcall start_game(int uMsg);
 void __cdecl free_game();
-bool __cdecl diablo_get_not_running();
-int __stdcall WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nShowCmd);
+BOOL __cdecl diablo_get_not_running();
+int PASCAL WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow);

```

Comment:
I've never seen the `PASCAL` calling convention being used in a C/C++ project before. Most often, `WINAPI` is used for the `WinMain` function.

Would `WinMain` still be bin exact with the following signature?

```cpp
int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow);
```

Reply:
I wasn't 100% sure what to use here, since `PASCAL/WINAPI/APIENTRY/CALLBACK` are all __stdcalls. Given that the demo version is a **direct** copy/paste of the WinMain routine from the DX2 SDK, this signature is exactly the same as what was in the SDK. So it should be correct, though I'd use APIENTRY for my code.

Reply:
Ok, makes sense. Still boggles my mind to have a PASCAL calling convention though, as the arguments are passed in reverse order as opposed to regular C calling conventions?

Reply:
After a bit of research it seems that `pascal` with `near/far` was common in 16-bit programs. When 32-bit was introduced it became obsolete. During the 16->32 transition, it was often necessary to keep compatibility with both, so DOS and Windows 95 could be targeted (similar to how most programs **still** have 32 and 64 versions to this day). Compilers adapted so that `PASCAL` evaluates to `__stdcall` for 32-bit, but continued to use traditional pascal for 16-bit. See here:
https://github.com/Alexpux/mingw-w64/blob/master/mingw-w64-tools/widl/include/windef.h#L165
To further complicate the issue, Diablo was written during the tail-end of the 16-bit era. I recall the 1994 or so pitch was that it would be a turn based-DOS game. They started development before both Windows 95 and DirectX 1.0 came out, and the first screenshots appeared around the time they did.

I don't think it should be an issue though, PASCAL and WINAPI generate the same code. It was likely also used for the WindowProc routine, i.e.:
`LONG FAR PASCAL WindowProc(HWND win, UINT msg, WPARAM wparam, LPARAM lparam)`

Reply:
I'll probably change it back, just to prevent confusion.

Reply:
Generally, I'd prefer if when converting a function from one call type to another, We could use an alias instead if just changing the call type, for the sake of documentation and future reference.  

for example, instead of doing 
```
__original_calltype --> __stdcall
```

I'd prefer  
```
#define ORIGINAL_CALLTYPE __stdcall
```

:D

Reply:
I seam to remember having heard that they originally wanted to target dos.